### PR TITLE
bpy: Added a new subcommand encode-dir to encode entire directories

### PR DESCRIPTION
**Pull request by Yoric**
This will simplify generating large batches of tests.

### CODE REVIEW

File: fbssdc/bpy.py [hunk 1]
```diff
@@ -7,15 +7,51 @@
...
+  types = idl.parse_es6_idl()
+  ty_script = types.interfaces['Script']
+  string_dict = strings.read_dict(dict_file, with_signature=True)
+  in_path = os.path.abspath(in_path)
+  out_path = os.path.abspath(out_path)
+  ignored_out_directory = tempfile.TemporaryDirectory()
+  for root, _, sources in os.walk(in_path):
+    # 1. Prepare destination directory
+    suffix = os.path.relpath(root, in_path)
+    print ("Root: {root}, suffix: {suffix}, out_path: {out_path}".format(root=root,suffix=suffix,out_path=out_path))
```
Comment (vdjeric) [inline fbssdc/bpy.py]: isn't this more of a debug statement?

File: fbssdc/bpy.py [hunk 2]
```diff
@@ -7,15 +7,51 @@
 
 import format
 import idl
+import io
```
Comment (vdjeric) [inline fbssdc/bpy.py]: do you use this module anywhere?

File: fbssdc/bpy.py [hunk 3]
```diff
@@ -7,15 +7,51 @@
...
+
+    for source in sources:
+      source_path=os.path.join(root, source)
+      if not source[-3:] == ".js":
+        print ("...skipping {}".format(source_path))
+        continue
+
+      # 2. Extract AST
+      print ("Preprocessing {}".format(source_path))
+      process = subprocess.run([binjs_encode, "--quiet", "--show-ast", "--in", source_path, "--out", ignored_out_directory.name], capture_output=True)
```
Comment (vdjeric) [inline fbssdc/bpy.py]: nit: you're a bit inconsistent about using single vs double quoted strings

File: fbssdc/bpy.py [hunk 4]
```diff
@@ -89,6 +125,23 @@ def main():
...
+
+  encode_full_parser = subs.add_parser('encode-dir', help='Encode a full directory (from .js).',
+                                  description='''Caveats:
+(1) dictionary misses are not supported yet
+(2) not all file sections are compressed natively yet;
+    the output should be compressed with Brotli
+''')
+  encode_full_parser.add_argument('binjs_encode',
+                             help='path to the binjs_encode binary')
+  encode_full_parser.add_argument('dictionary', type=argparse.FileType('rb'),
```
Comment (vdjeric) [inline fbssdc/bpy.py]: what do you think of making dictionary arg optional so that it can default to an empty external dictionary

Comment (Yoric) [reply fbssdc/bpy.py]: Maybe in a followup bug. For the moment, I don't think that `format.write` supports dictionary as an optional argument.

**Pull request merged**